src/examples/allocator/ServerAllocator.sol

```diff
@@ -0,0 +1,260 @@
+        if (attests_.length != nonces_.length) {
+            revert InvalidInput();
+        }
+        _consumeNonces(nonces_, attests_);
+    }
+
+    /// @inheritdoc IServerAllocator
+    function consumeViaSignature(NonceConsumption calldata data_, bytes calldata signature_) external {
+        if (data_.attests.length != data_.nonces.length) {
+            revert InvalidInput();
+        }
+        address signer = _validateNonceConsumption(data_, signature_);
+        if (signer != data_.signer || !_containsSigner(signer)) {
+            // first check is optional, can be deleted for gas efficiency
+            revert InvalidSignature(signature_, signer);
+        }
+        _consumeNonces(data_.nonces, data_.attests);
+    }
+
+    /// @inheritdoc IERC1271
+    function isValidSignature(bytes32 hash_, bytes calldata signature_) external view returns (bytes4 magicValue) {
+        address signer = _validateSignedHash(hash_, signature_);
+        if (!_containsSigner(signer)) {
+            revert InvalidSignature(signature_, signer);
+        }
+        return IERC1271.isValidSignature.selector;
+    }
+
+    /// @inheritdoc IServerAllocator
+    function checkIfSigner(address signer_) external view returns (bool) {
+        return _containsSigner(signer_);
+    }
+
+    /// @inheritdoc IServerAllocator
+    function getAllSigners() external view returns (address[] memory) {
+        return _activeSigners;
+    }
+
+    /// @inheritdoc IServerAllocator
+    function checkAttestExpirations(bytes32 attest_) external view returns (uint256[] memory) {
+        return _checkAttestExpirations(attest_);
+    }
+
+    /// @inheritdoc IServerAllocator
+    function checkAttestExpirations(address sponsor_, uint256 id_, uint256 amount_) external view returns (uint256[] memory) {
+        return _checkAttestExpirations(keccak256(abi.encode(sponsor_, id_, amount_)));
+    }
+
+    /// @inheritdoc IServerAllocator
+    function getCompactContract() external view returns (address) {
+        return _COMPACT_CONTRACT;
+    }
+
+    function _registerAttest(bytes32 attest_, uint256 expiration_) internal {
+        if (expiration_ < block.timestamp) {
+            revert Expired(expiration_, block.timestamp);
+        }
+        uint256 count = ++_attestCounts[attest_];
+        bytes32 countedAttest = keccak256(abi.encode(attest_, count));
+
+        _attestExpirations[countedAttest] = expiration_;
+
+        emit AttestRegistered(attest_, expiration_);
+    }
+
+    /// Todo: This will lead to always the last registered hash being consumed.
+    function _consumeNonces(uint256[] calldata nonces_, bytes32[] calldata attests_) internal {
+        ITheCompact(_COMPACT_CONTRACT).consume(nonces_);
+        uint256 nonceLength = attests_.length;
+        for (uint256 i = 0; i < nonceLength; ++i) {
+            bytes32 hashToConsume = attests_[i];
+            if (hashToConsume != bytes32(0)) {
```

Comment:
when would a bytes32(0) value be supplied here?

Reply:
It can be supplied in case there was not previously an attest registered for that specific nonce